drivers: i2c: Add an i2c_configure_dt function

rruuaanng opened the pull request:
I prefer using a format like `gpio_pin_configure_dt(&spec, ...)` rather than `i2c_configure(spec.bus, ...)`. I believe I'm not the only one who has thought about this.

teburd commented on include/zephyr/drivers/i2c.h:
```diff
@@ -722,6 +722,27 @@ static inline int z_impl_i2c_configure(const struct device *dev,
+ *
+ * @return a value from i2c_configure()
+ */
+__syscall int i2c_configure_dt(const struct i2c_dt_spec *spec, uint32_t dev_config);
```
Doesn't need to be a syscall, can be a simple inline wrapper around i2c_configure, otherwise looks ok to me

teburd reviewed (changes requested):
Won't make 4.1, this isn't a bug fix, but should be in for 4.2

rruuaanng committed:
drivers: i2c: Add an i2c_configure_dt function

Add an i2c_configure_dt function, that is equivalent
to calling i2c_configure(spec->bus, dev_config) to
simplify the calling.

Signed-off-by: James Roy <rruuaanng@outlook.com>

rruuaanng committed:
doc: releases: Add a i2c release entry

Add the i2c_configure_dt function in the i2c
subsystem to simplify calls through spec by
eliminating unnecessary member references.

Signed-off-by: James Roy <rruuaanng@outlook.com>
